packages/core/src/tracing/trace.ts

```diff
@@ -37,43 +40,18 @@ export function trace<T>(
 
   scope.setSpan(activeSpan);
 
-  function finishAndSetSpan(): void {
-    activeSpan && activeSpan.end();
-    scope.setSpan(parentSpan);
-  }
-
-  let maybePromiseResult: T;
-  try {
-    maybePromiseResult = callback(activeSpan);
-  } catch (e) {
-    activeSpan && activeSpan.setStatus('internal_error');
-    onError(e, activeSpan);
-    finishAndSetSpan();
-    afterFinish();
-    throw e;
-  }
-
-  if (isThenable(maybePromiseResult)) {
-    // @ts-expect-error - the isThenable check returns the "wrong" type here
-    return maybePromiseResult.then(
```

Comment:
Hmm, I see some nextjs E2E tests are failing - can it be that this was "swallowed" before due to this (which we changed for `startSpan` but I think not for `trace`? 🤔  cc @Lms24  / @lforst

Reply:
I think the failures might be legit. I don't think there was anything being swallowed. It seems like we're timing out on waiting for elements to appear on screen. Can you verify that you are returning all the values and not stalling anything? Are we returning the response here? https://github.com/getsentry/sentry-javascript/pull/10012/files#diff-6a177c0f974e405d7e55f4d0155181376d92b37749269c0904cdc6e94b17ef84L97

Reply:
I added it back. Let's see.

Reply:
OMG. good tests!

Reply:
Seems like an oversight a linter should have caught...

Reply:
Re added the lint rule here https://github.com/getsentry/sentry-javascript/pull/10049